generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/execution/FunctionDataFetcher.kt

```diff
@@ -120,12 +120,15 @@ open class FunctionDataFetcher(
         }
 
     /**
-     * Once all parameters values are properly converted, this function will be called to run a simple blocking function.
-     * If you need to override the exception handling you can override this method.
+     * Once all parameter values are properly converted, this function will be called to run a simple blocking function.
+     * If you need to override the exception handling, you can override this method.
      */
-    protected open fun runBlockingFunction(parameterValues: Map<KParameter, Any?>): Any? = try {
-        fn.callBy(parameterValues)
-    } catch (exception: InvocationTargetException) {
-        throw exception.cause ?: exception
-    }
+    protected open fun runBlockingFunction(parameterValues: Map<KParameter, Any?>): CompletableFuture<Any?>  =
```

Comment:
This will cause an allocation of an object of type `CompletableFuture` for fields that are already materialized (in-memory).

Reply:
I made the `CompletableFuture` wrap the success-case for consistency, but could also peel that back as it's really only the `throw ...` case that is the problem and inconsistent with the equivalent behavior for `suspend` resolvers.

Shall I do that?